integration/tests/epochs/recover_epoch/recover_epoch_efm_test.go

```diff
@@ -19,16 +23,79 @@ type RecoverEpochSuite struct {
 	Suite
 }
 
-// TestRecoverEpoch ensures that the recover_epoch transaction flow works as expected. This test will simulate the network going
-// into EFM by taking a consensus node offline before completing the DKG. While in EFM mode the test will execute the efm-recover-tx-args
-// CLI command to generate transaction arguments to submit a recover_epoch transaction, after submitting the transaction the test will
-// ensure the network is healthy.
+// TestRecoverEpoch ensures that the recover epoch governance transaction flow works as expected and a network that
+// enters Epoch Fallback Mode can successfully recover. This test will do the following:
+// 1. Manually triggers EFM by turning off a collection node before the end of the DKG forcing the DKG to fail.
+// 2. Generates epoch recover transaction args using the epoch efm-recover-tx-args.
+// 3. Submit recover epoch transaction.
+// 4. Ensure expected EpochRecover event is emitted.
+// Currently, this test does not test the processing of the EpochRecover event see this issue: https://github.com/onflow/flow-go/issues/6164
 func (s *RecoverEpochSuite) TestRecoverEpoch() {
-	s.AwaitEpochPhase(context.Background(), 0, flow.EpochPhaseSetup, 20*time.Second, time.Second)
-	fmt.Println("in epoch phase setup")
+	// 1. Manually trigger EFM
+	// wait until the epoch setup phase to force network into EFM
+	s.AwaitEpochPhase(s.Ctx, 0, flow.EpochPhaseSetup, 10*time.Second, 500*time.Millisecond)
+	// pausing consensus node will force the network into EFM
+	ln := s.GetContainersByRole(flow.RoleCollection)[0]
+	_ = ln.Pause()
+	s.AwaitFinalizedView(s.Ctx, 32, 2*time.Minute, 500*time.Millisecond)
+	// start the paused execution node now that we are in EFM
+	require.NoError(s.T(), ln.Start())
 
-	sns := s.GetContainersByRole(flow.RoleConsensus)
-	_ = sns[0].Pause()
+	// get the latest snapshot and start new container with it
+	epoch1FinalView, err := s.Net.BootstrapSnapshot.Epochs().Current().FinalView()
+	require.NoError(s.T(), err)
 
-	// @TODO: trigger EFM manually
+	// wait for at least the first block of the next epoch to be sealed so that we can
+	// ensure that we are still in the same epoch after the final view of that epoch indicating we are in EFM
+	s.TimedLogf("waiting for epoch transition (finalized view %d) before pausing container", epoch1FinalView+1)
+	s.AwaitFinalizedView(s.Ctx, epoch1FinalView+1, 2*time.Minute, 500*time.Millisecond)
+	s.TimedLogf("observed finalized view %d -> pausing container", epoch1FinalView+1)
+
+	//assert transition to second epoch did not happen
+	//if counter is still 0, epoch emergency fallback was triggered as expected
+	s.AssertInEpoch(s.Ctx, 0)
+
+	// 2. Generate transaction arguments for epoch recover transaction.
+	// generate epoch recover transaction args
+	collectionClusters := uint64(1)
```

Comment:
Shall we take some of the data from root epoch setup.

Reply:
https://github.com/onflow/flow-go/pull/6156/commits/777267fff4dedeebd234b180ddba59327e95c206